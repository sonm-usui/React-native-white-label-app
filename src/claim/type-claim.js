// @flow

import type {
  CustomError,
  InitialTestAction,
  ResetAction,
  NotificationPayload,
} from '../common/type-common'

export const CLAIM_STORAGE_SUCCESS = 'CLAIM_STORAGE_SUCCESS'
export type ClaimStorageSuccessAction = {
  type: typeof CLAIM_STORAGE_SUCCESS,
  messageId: string,
  claimId: string,
  issueDate: number,
}

export const CLAIM_STORAGE_FAIL = 'CLAIM_STORAGE_FAIL'
export type ClaimStorageFailAction = {
  type: typeof CLAIM_STORAGE_FAIL,
  messageId: string,
  error: CustomError,
}

export const MAP_CLAIM_TO_SENDER = 'MAP_CLAIM_TO_SENDER'
export type MapClaimToSenderAction = {
  type: typeof MAP_CLAIM_TO_SENDER,
  claimUuid: string,
  senderDID: string,
  myPairwiseDID: string,
  logoUrl: string,
  issueDate: number,
  name: string,
  senderName?: string,
}

export const HYDRATE_CLAIM_MAP = 'HYDRATE_CLAIM_MAP'
export type HydrateClaimMapAction = {
  type: typeof HYDRATE_CLAIM_MAP,
  claimMap: ClaimMap,
}

export const HYDRATE_CLAIM_MAP_FAIL = 'HYDRATE_CLAIM_MAP_FAIL'
export type HydrateClaimMapFailAction = {
  type: typeof HYDRATE_CLAIM_MAP_FAIL,
  error: CustomError,
}

export type ClaimPushPayload = {
  connectionHandle: number,
}
export type Claim = NotificationPayload & ClaimPushPayload

export const CLAIM_RECEIVED = 'CLAIM_RECEIVED'
export type ClaimReceivedAction = {
  type: typeof CLAIM_RECEIVED,
  claim: Claim,
}

export type ClaimAction =
  | ClaimStorageSuccessAction
  | ClaimStorageFailAction
  | MapClaimToSenderAction
  | HydrateClaimMapAction
  | HydrateClaimMapFailAction
  | InitialTestAction
  | ResetAction
  | ClaimReceivedAction

export type ClaimMap = {
  +[claimUuid: string]: {
    senderDID: string,
    myPairwiseDID: string,
    logoUrl: string,
    issueDate: number,
    name?: string,
    senderName?: string,
  },
}

export type ClaimStore = {
  +[string]: {
    error?: CustomError,
  },
  claimMap: ClaimMap,
}

export const ERROR_CLAIM_HYDRATE_FAIL = {
  message: 'Failed to hydrate claim map',
  code: 'CL-001',
}

export type ClaimInfo = {
  referent: string,
  schema_id: string,
  attrs: {
    [attrName: string] : string
  },
}
