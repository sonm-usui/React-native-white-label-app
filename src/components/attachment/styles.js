// @flow
import { StyleSheet, Platform } from 'react-native'
import { verticalScale, moderateScale } from 'react-native-size-matters'
import { colors, fontSizes, fontFamily } from '../../common/styles/constant'

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
  },
  scrollViewWrapper: {
    backgroundColor: colors.white,
    paddingLeft: '5%',
    paddingRight: '5%',
  },
  wrapper: {
    backgroundColor: colors.white,
    paddingTop: moderateScale(12),
    ...Platform.select({
      ios: {
        borderBottomColor: colors.gray5,
        borderBottomWidth: StyleSheet.hairlineWidth,
      },
      android: {
        borderBottomColor: colors.gray5,
        borderBottomWidth: 1,
      },
    }),
  },
  title: {
    fontSize: verticalScale(fontSizes.size6),
    color: colors.gray3,
    width: '100%',
    textAlign: 'left',
    marginBottom: moderateScale(2),
    fontFamily: fontFamily,
    lineHeight: verticalScale(17),
  },
  content: {
    fontSize: verticalScale(fontSizes.size3),
    fontWeight: '700',
    color: '#505050',
    width: '100%',
    textAlign: 'left',
    fontFamily: fontFamily,
    lineHeight: verticalScale(23),
  },
  contentGray: {
    fontSize: verticalScale(fontSizes.size5),
    fontWeight: '700',
    color: colors.gray1,
    width: '100%',
    textAlign: 'left',
    fontFamily: fontFamily,
  },
  textAvatarWrapper: {
    width: '98.5%',
    flexDirection: 'row',
  },
  textWrapper: {
    width: '90%',
    paddingBottom: moderateScale(10),
  },
  avatarWrapper: {
    width: '15%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  photoAttachment: {
    width: moderateScale(150),
    height: moderateScale(150),
  },
  attachmentIcon: {
    width: moderateScale(64),
    height: moderateScale(64),
  },
  nameIconContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  attachmentNameContainer: {
    flex: 1,
    marginLeft: moderateScale(10),
  },
  parentWrapper: {
    flexDirection: 'row',
  },
  svgStyles: {
    marginRight: moderateScale(16),
  },
  extensionNameStyle: {
    fontSize: verticalScale(fontSizes.size6),
    fontWeight: '600',
    color: colors.gray1,
    width: '100%',
    textAlign: 'left',
    fontFamily: fontFamily,
  },
})
