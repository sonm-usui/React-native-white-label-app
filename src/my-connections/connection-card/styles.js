// @flow
import { StyleSheet, Platform } from 'react-native'
import { colors, fontFamily, fontSizes } from '../../common/styles/constant'
import { verticalScale, moderateScale } from 'react-native-size-matters'

const styles = StyleSheet.create({
  animatedContainer: {
    position: 'absolute',
    zIndex: 1000,
    width: '100%',
    height: moderateScale(90, 0.25),
    alignItems: 'center',
    justifyContent: 'center',
  },
  itemContainer: {
    width: '47%',
    minHeight: moderateScale(120, 2),
    marginRight: 20,
    marginBottom: 35,
    borderRadius: 20,
    shadowOffset: { width: 0, height: 2.5 },
    shadowColor: colors.black,
    shadowOpacity: Platform.OS === 'android' ? 0.05 : 0.1,
    elevation: Platform.OS === 'android' ? 10 : 20,
    backgroundColor: colors.white,
    paddingVertical: verticalScale(16),
    display: 'flex',
    justifyContent: 'center',
  },
  itemFailed: {
    borderWidth: 2,
    borderColor: colors.red,
  },
  initialsContainer: {
    width: 32,
    height: 32,
    borderRadius: 16,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: colors.gray5,
  },
  initialsText: {
    fontFamily: fontFamily,
    fontSize: verticalScale(fontSizes.size5),
    fontWeight: 'bold',
    color: colors.gray1,
  },
  avatarSection: {
    alignItems: 'center',
    borderRadius: 50,
  },
  dateButtonSection: {
    width: '32%',
    height: '100%',
    flexDirection: 'row',
    alignItems: 'flex-end',
  },
  buttonSection: {
    height: '50%',
    width: '30%',
    justifyContent: 'flex-end',
  },
  companyNameText: {
    paddingLeft: '5%',
    paddingRight: '5%',
    marginTop: 5,
    fontFamily: fontFamily,
    fontSize: verticalScale(fontSizes.size4),
    fontWeight: '400',
    color: colors.gray1,
    textAlign: 'center',
  },
  newButtonSection: {
    height: '100%',
    width: '30%',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  newLabel: {
    width: moderateScale(64),
    height: verticalScale(16),
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.main,
    marginBottom: 5,
  },
  newLabelText: {
    fontFamily: fontFamily,
    fontSize: verticalScale(fontSizes.size9),
    fontWeight: '500',
    color: colors.white,
  },
  outerContainer: {
    position: 'absolute',
    zIndex: 1000,
    width: '100%',
    height: moderateScale(90, 0.25),
    alignItems: 'center',
    justifyContent: 'center',
  },
  avatarStyle: {
    width: 80,
    height: 80,
    borderRadius: 100,
  },
  customGreenBadgeContainer: {
    position: 'absolute',
    top: verticalScale(10),
    right: verticalScale(10),
    width: moderateScale(22),
    height: moderateScale(22),
    borderRadius: moderateScale(22) / 2,
    backgroundColor: colors.main,
    alignItems: 'center',
    justifyContent: 'center',
  },
  numberText: {
    fontFamily: fontFamily,
    fontSize: moderateScale(14),
    fontWeight: '500',
    color: colors.white,
  },
  errorImage: {
    position: 'absolute',
    right: 10,
    top: 10,
    width: 24,
    height: 24,
  },
  loader: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1
  }
})

export { styles }
