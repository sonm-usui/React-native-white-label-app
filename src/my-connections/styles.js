// @flow
import { StyleSheet } from 'react-native'
import { colors } from '../common/styles/constant'
import { verticalScale, moderateScale } from 'react-native-size-matters'

const externalStyles = StyleSheet.create({
  outerContainer: {
    flex: 1,
  },
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: colors.white,
    flex: 1,
  },
  flatListContainer: {
    width: '100%',
    height: '100%',
    backgroundColor: colors.white,
  },
  flatListInnerContainer: {
    paddingBottom: moderateScale(170, 0.25),
    margin: 15,
    padding: 10,
    marginBottom: 40,
  },
  blurContainer: {
    position: 'absolute',
    top: 0,
    width: '100%',
    height: verticalScale(90),
  },
  backgroundImage: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export { externalStyles }
