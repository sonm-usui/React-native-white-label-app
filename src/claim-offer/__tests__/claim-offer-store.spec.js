// @flow
import claimOfferStore, {
  claimOfferReceived,
  claimOfferShown,
  claimOfferIgnored,
  claimOfferRejected,
  sendClaimRequest,
  claimRequestSuccess,
  claimRequestFail,
  acceptClaimOffer,
  addSerializedClaimOffer,
  hydrateClaimOffers,
  saveClaimOffersSaga,
  hydrateClaimOffersSaga,
  saveSerializedClaimOffer,
  claimOfferAccepted,
  claimOfferShowStart,
  resetClaimRequestStatus,
  getColorTheme,
} from '../claim-offer-store'
import {
  CLAIM_OFFERS,
  SAVE_CLAIM_OFFERS_SUCCESS,
  SAVE_CLAIM_OFFERS_FAIL,
  ERROR_SAVE_CLAIM_OFFERS,
} from '../type-claim-offer'
import { getConnectionHistory } from '../../store/store-selector'
import {
  serializeClaimOffer,
  getHandleBySerializedConnection,
  getClaimHandleBySerializedClaimOffer,
  getClaimOfferState,
  sendClaimRequest as sendClaimRequestApi,
} from '../../bridge/react-native-cxs/RNCxs'
import { CLAIM_STORAGE_SUCCESS } from '../../claim/type-claim'
import {
  claimOffer,
  claimOfferId as uid,
  claimOfferIssueDate as issueDate,
  pairwiseConnection,
  claimOfferPayload,
  serializedClaimOffers,
  serializedClaimOffer,
  vcxSerializedConnection,
  connectionHistory,
  colorTheme,
  claimUUID,
  caseInsensitiveAttributes,
} from '../../../__mocks__/static-data'
import { expectSaga } from 'redux-saga-test-plan'
import * as matchers from 'redux-saga-test-plan/matchers'
import { throwError } from 'redux-saga-test-plan/providers'
import { secureSet, getHydrationItem } from '../../services/storage'
import {
  VCX_INIT_POOL_SUCCESS,
  VCX_INIT_SUCCESS,
} from '../../store/type-config-store'
import { colors } from '../../common/styles'

describe('claim offer store', () => {
  const initialAction = { type: 'INITIAL_TEST_ACTION' }
  let initialState = {
    vcxSerializedClaimOffers: {},
  }
  let newState = {
    vcxSerializedClaimOffers: {},
  }
  const claimOfferVcxInitialState = 1

  beforeEach(() => {
    initialState = claimOfferStore(undefined, initialAction)
  })

  it('should correctly calculate initial state', () => {
    expect(initialState).toMatchSnapshot()
  })

  it('claim offer is received', () => {
    newState = claimOfferStore(
      initialState,
      claimOfferReceived(claimOffer.payload, claimOffer.payloadInfo)
    )
    expect(newState).toMatchSnapshot()
  })

  it('claim offer is shown', () => {
    newState = claimOfferStore(newState, claimOfferShown(uid))
    expect(newState).toMatchSnapshot()
  })

  it('claim offer is ignored', () => {
    newState = claimOfferStore(newState, claimOfferIgnored(uid))
    expect(newState).toMatchSnapshot()
  })

  it('claim offer is rejected', () => {
    newState = claimOfferStore(newState, claimOfferRejected(uid))
    expect(newState).toMatchSnapshot()
  })

  it('claim offer is accepted', () => {
    newState = claimOfferStore(
      newState,
      acceptClaimOffer(uid, claimOffer.payload.issuer.did)
    )
    expect(newState).toMatchSnapshot()
  })

  it('claim request is sent', () => {
    newState = claimOfferStore(
      newState,
      sendClaimRequest(uid, claimOfferPayload)
    )
    expect(newState).toMatchSnapshot()
  })

  it('claim request is success', () => {
    newState = claimOfferStore(
      newState,
      claimRequestSuccess(
        uid,
        issueDate,
        colorTheme,
        claimUUID,
        caseInsensitiveAttributes
      )
    )
    expect(newState).toMatchSnapshot()
  })

  it('claim request is fail', () => {
    newState = claimOfferStore(
      newState,
      claimRequestFail(uid, {
        code: 'TEST-100',
        message: 'Claim request failed',
      })
    )
    expect(newState).toMatchSnapshot()
  })

  it('action: ADD_SERIALIZED_CLAIM_OFFER', () => {
    newState = claimOfferStore(
      newState,
      addSerializedClaimOffer(
        serializedClaimOffer,
        pairwiseConnection.identifier,
        uid,
        claimOfferVcxInitialState
      )
    )
    expect(newState).toMatchSnapshot()
  })

  it('action: CLAIM_OFFER_SHOW_START', () => {
    expect(
      claimOfferStore(newState, claimOfferShowStart(uid))
    ).toMatchSnapshot()
  })

  it('action: RESET_CLAIM_REQUEST_STATUS', () => {
    const stateAfterClaimRequestFail = claimOfferStore(
      newState,
      claimRequestFail(uid, {
        code: 'TEST-100',
        message: 'Claim request failed',
      })
    )
    expect(
      claimOfferStore(stateAfterClaimRequestFail, resetClaimRequestStatus(uid))
    ).toMatchSnapshot()
  })

  it('should reset claim offer store, if RESET action is raised', () => {
    expect(claimOfferStore(newState, { type: 'RESET' })).toMatchSnapshot()
  })

  it('action: HYDRATE_CLAIM_OFFERS_SUCCESS', () => {
    newState = claimOfferStore(
      newState,
      hydrateClaimOffers(JSON.parse(serializedClaimOffers))
    )
    expect(newState).toMatchSnapshot()
  })

  it('saga: saveSerializedClaimOfferSaga, success', () => {
    return expectSaga(
      saveClaimOffersSaga,
      addSerializedClaimOffer(
        serializedClaimOffer,
        pairwiseConnection.identifier,
        uid,
        claimOfferVcxInitialState
      )
    )
      .withState({ claimOffer: { vcxSerializedClaimOffers: {} } })
      .call(secureSet, CLAIM_OFFERS, '{"vcxSerializedClaimOffers":{}}')
      .put({ type: SAVE_CLAIM_OFFERS_SUCCESS })
      .run()
  })

  it('saga: saveSerializedClaimOfferSaga, fail', () => {
    const errorMessage = 'test error'
    const failSaveSerializedClaimOffers = new Error(errorMessage)

    return expectSaga(
      saveClaimOffersSaga,
      addSerializedClaimOffer(
        serializedClaimOffer,
        pairwiseConnection.identifier,
        uid,
        claimOfferVcxInitialState
      )
    )
      .withState({ claimOffer: { vcxSerializedClaimOffers: {} } })
      .provide([
        [
          matchers.call.fn(secureSet, CLAIM_OFFERS, '{}'),
          throwError(failSaveSerializedClaimOffers),
        ],
      ])
      .put({
        type: SAVE_CLAIM_OFFERS_FAIL,
        error: ERROR_SAVE_CLAIM_OFFERS(errorMessage),
      })
      .run()
  })

  it('saga: hydrateClaimOffersSaga, success', () => {
    const history = {
      data: {
        connections: {
          pairwiseIdentifier1: {
            data: connectionHistory['September 2017'].data,
          },
        },
      },
    }
    const data = {
      pairwiseIdentifier1: {
        uid: 'uid',
        remotePairwiseDID: 'remotePairwiseDID',
        senderLogoUrl: 'senderLogoUrl',
        claimId: claimUUID,
        caseInsensitiveAttributes: caseInsensitiveAttributes,
      },
    }

    const expected = {
      pairwiseIdentifier1: {
        ...data.pairwiseIdentifier1,
        colorTheme: colors.main,
      },
    }
    const expectedJSON = JSON.stringify(expected)

    return expectSaga(hydrateClaimOffersSaga)
      .provide([
        [
          matchers.call.fn(getHydrationItem, CLAIM_OFFERS),
          JSON.stringify(data),
        ],
        [matchers.select.selector(getConnectionHistory), history],
        [matchers.call.fn(getColorTheme, 'senderLogoUrl'), colors.main],
      ])
      .put(hydrateClaimOffers(JSON.parse(expectedJSON)))
      .run()
  })

  xit('saga: claimOfferAccepted, success', () => {
    const claimOfferPayload = {
      ...claimOffer.payload,
      ...claimOffer.payloadInfo,
    }
    const userDID = pairwiseConnection.identifier
    const claimOfferSerialized = {
      messageId: uid,
      serialized: serializedClaimOffer,
      state: 1,
    }
    const stateWithClaimOfferAndSerialized = {
      claimOffer: {
        [uid]: claimOfferPayload,
        vcxSerializedClaimOffers: {
          [userDID]: {
            [uid]: claimOfferSerialized,
          },
        },
      },
      connections: {
        data: {
          [userDID]: {
            ...pairwiseConnection,
            senderDID: claimOfferPayload.remotePairwiseDID,
            vcxSerializedConnection: vcxSerializedConnection,
          },
        },
      },
      config: {
        vcxInitializationState: VCX_INIT_SUCCESS,
        vcxPoolInitializationState: VCX_INIT_POOL_SUCCESS,
      },
    }
    const claimHandle = 1
    const connectionHandle = 1
    const paymentHandle = 0

    return expectSaga(
      claimOfferAccepted,
      acceptClaimOffer(uid, claimOfferPayload.issuer.did)
    )
      .withState(stateWithClaimOfferAndSerialized)
      .provide([
        [
          matchers.call.fn(
            getHandleBySerializedConnection,
            vcxSerializedConnection
          ),
          connectionHandle,
        ],
        [
          matchers.call.fn(
            getClaimHandleBySerializedClaimOffer,
            claimOfferSerialized
          ),
          claimHandle,
        ],
        [
          matchers.call.fn(
            sendClaimRequestApi,
            claimHandle,
            connectionHandle,
            paymentHandle
          ),
          true,
        ],
      ])
      .dispatch({ type: CLAIM_STORAGE_SUCCESS, messageId: uid })
      .call(sendClaimRequestApi, claimHandle, connectionHandle, paymentHandle)
      .call(
        saveSerializedClaimOffer,
        claimHandle,
        pairwiseConnection.identifier,
        uid
      )
      .put(sendClaimRequest(uid, claimOfferPayload))
      .run()
  })

  it('saga: saveSerializedClaimOffer', () => {
    const claimHandle = 1
    return expectSaga(
      saveSerializedClaimOffer,
      claimHandle,
      pairwiseConnection.identifier,
      uid
    )
      .provide([
        [
          matchers.call.fn(serializeClaimOffer, claimHandle),
          serializedClaimOffer,
        ],
        [
          matchers.call.fn(getClaimOfferState, claimHandle),
          claimOfferVcxInitialState,
        ],
      ])
      .put(
        addSerializedClaimOffer(
          serializedClaimOffer,
          pairwiseConnection.identifier,
          uid,
          claimOfferVcxInitialState
        )
      )
      .call(serializeClaimOffer, claimHandle)
      .call(getClaimOfferState, claimHandle)
      .run()
  })
})
