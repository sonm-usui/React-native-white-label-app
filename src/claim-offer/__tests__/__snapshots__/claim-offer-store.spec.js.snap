// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`claim offer store action: ADD_SERIALIZED_CLAIM_OFFER 1`] = `
Object {
  "usd123": Object {
    "caseInsensitiveAttributes": Array [
      Object {
        "address1": "Address 1",
        "address2": "Address 2",
      },
    ],
    "claimId": "claimUUID",
    "claimRequestStatus": "CLAIM_REQUEST_FAIL",
    "colorTheme": "#86B93B",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issueDate": 123456789,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "FAILED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {
    "pairwiseIdentifier1": Object {
      "usd123": Object {
        "messageId": "usd123",
        "serialized": "{}",
        "state": 1,
      },
    },
  },
}
`;

exports[`claim offer store action: CLAIM_OFFER_SHOW_START 1`] = `
Object {
  "usd123": Object {
    "caseInsensitiveAttributes": Array [
      Object {
        "address1": "Address 1",
        "address2": "Address 2",
      },
    ],
    "claimId": "claimUUID",
    "claimRequestStatus": "NONE",
    "colorTheme": "#86B93B",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issueDate": 123456789,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "RECEIVED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {
    "pairwiseIdentifier1": Object {
      "usd123": Object {
        "messageId": "usd123",
        "serialized": "{}",
        "state": 1,
      },
    },
  },
}
`;

exports[`claim offer store action: HYDRATE_CLAIM_OFFERS_SUCCESS 1`] = `
Object {
  "pairwiseIdentifier1": Object {
    "uid": "{}",
  },
}
`;

exports[`claim offer store action: RESET_CLAIM_REQUEST_STATUS 1`] = `
Object {
  "usd123": Object {
    "caseInsensitiveAttributes": Array [
      Object {
        "address1": "Address 1",
        "address2": "Address 2",
      },
    ],
    "claimId": "claimUUID",
    "claimRequestStatus": "NONE",
    "colorTheme": "#86B93B",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issueDate": 123456789,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "RECEIVED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {
    "pairwiseIdentifier1": Object {
      "usd123": Object {
        "messageId": "usd123",
        "serialized": "{}",
        "state": 1,
      },
    },
  },
}
`;

exports[`claim offer store claim offer is accepted 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "NONE",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "ACCEPTED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim offer is ignored 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "NONE",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim offer is received 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "NONE",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "RECEIVED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim offer is rejected 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "NONE",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "REJECTED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim offer is shown 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "NONE",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "SHOWN",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim request is fail 1`] = `
Object {
  "usd123": Object {
    "caseInsensitiveAttributes": Array [
      Object {
        "address1": "Address 1",
        "address2": "Address 2",
      },
    ],
    "claimId": "claimUUID",
    "claimRequestStatus": "CLAIM_REQUEST_FAIL",
    "colorTheme": "#86B93B",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issueDate": 123456789,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "FAILED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim request is sent 1`] = `
Object {
  "usd123": Object {
    "claimRequestStatus": "SENDING_CLAIM_REQUEST",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "ACCEPTED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store claim request is success 1`] = `
Object {
  "usd123": Object {
    "caseInsensitiveAttributes": Array [
      Object {
        "address1": "Address 1",
        "address2": "Address 2",
      },
    ],
    "claimId": "claimUUID",
    "claimRequestStatus": "CLAIM_REQUEST_SUCCESS",
    "colorTheme": "#86B93B",
    "data": Object {
      "claimDefinitionSchemaSequenceNumber": 36,
      "name": "Home Address",
      "revealedAttributes": Array [
        Object {
          "data": "Address Address Address",
          "label": "Address 1",
        },
        Object {
          "data": "Address 2 Address 2 Address 2",
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "error": null,
    "issueDate": 123456789,
    "issuer": Object {
      "did": "issuerDid",
      "name": "Test Issuer",
    },
    "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
    "senderLogoUrl": "http://testissuer.com/logoUrl.png",
    "status": "ISSUED",
    "statusMsg": "pending",
    "uid": "usd123",
  },
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store should correctly calculate initial state 1`] = `
Object {
  "vcxSerializedClaimOffers": Object {},
}
`;

exports[`claim offer store should reset claim offer store, if RESET action is raised 1`] = `
Object {
  "vcxSerializedClaimOffers": Object {},
}
`;
